src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -37,6 +38,16 @@ public sealed class AnalyzerConfigSet
         private readonly ConcurrentDictionary<ReadOnlyMemory<char>, string> _diagnosticIdCache =
             new ConcurrentDictionary<ReadOnlyMemory<char>, string>(CharMemoryEqualityComparer.Instance);
 
+        // PERF: Most files will probably have the same options, so share the dictionary instances
+        private readonly ConcurrentCache<ImmutableArray<Section>, AnalyzerConfigOptionsResult> _optionsCache =
+            new ConcurrentCache<ImmutableArray<Section>, AnalyzerConfigOptionsResult>(50, SequenceEqualComparer<Section>.Instance); // arbitrary size
+
+        private readonly ObjectPool<TreeOptions.Builder> _treeOptionsPool =
+            new ObjectPool<TreeOptions.Builder>(() => ImmutableDictionary.CreateBuilder<string, ReportDiagnostic>(CaseInsensitiveComparison.Comparer));
```

Comment:
Why this comparer vs. the named comparers that already exist in this section of the code?